core/query/src/main/java/datawave/core/query/logic/DelegatingQueryLogic.java

```diff
@@ -353,6 +323,51 @@ public void validate(Map<String,List<String>> parameters) throws IllegalArgument
         delegate.validate(parameters);
     }
 
+    @Override
+    public int getMaxConcurrentTasks() {
+        return delegate.getMaxConcurrentTasks();
+    }
+
+    @Override
+    public void setMaxConcurrentTasks(int maxConcurrentTasks) {
+        delegate.setMaxConcurrentTasks(maxConcurrentTasks);
+    }
+
+    @Override
+    public boolean canRunQuery(Collection<String> userRoles) {
+        return delegate.canRunQuery(userRoles);
+    }
+
+    @Override
+    public void setRequiredRoles(Set<String> requiredRoles) {
+        delegate.setRequiredRoles(requiredRoles);
+    }
+
+    @Override
+    public Set<String> getRequiredRoles() {
+        return delegate.getRequiredRoles();
+    }
+
+    @Override
+    public ProxiedUserDetails getCurrentUser() {
+        return delegate.getCurrentUser();
+    }
+
+    @Override
+    public void setCurrentUser(ProxiedUserDetails currentUser) {
+        delegate.setCurrentUser(currentUser);
+    }
+
+    @Override
+    public ProxiedUserDetails getServerUser() {
+        return delegate.getServerUser();
+    }
+
+    @Override
+    public void setServerUser(ProxiedUserDetails serverUser) {
+        delegate.setServerUser(serverUser);
+    }
+
```

Comment:
after seeing identical code in the BaseQueryLogic maybe this should be pulled out into a separate class and composed?

Reply:
I don't understand your comment.  BaseQueryLogic isn't doing any kind of delegation like this class is doing.